src/Castle.Windsor/MicroKernel/SubSystems/Conversion/TypeNameConverter.cs

```diff
@@ -144,6 +149,29 @@ private bool InitializeAppDomainAssemblies(bool forceLoad)
 					Scan(assembly);
 				}
 			}
+#else
+			if (assemblies.Count == 0)
+			{
+				var context = DependencyContext.Default;
```

Comment:
@alinapopa asked me to help clarify some questions about DependencyModel.  I am more or less the maintainer of it - see [here](https://github.com/dotnet/core-setup/tree/master/src/managed/Microsoft.Extensions.DependencyModel) for the latest code and issues, etc.

I'm not really going to recommend whether you use it one way or the other, but I just wanted to give information and clear up some questions.


To start, my 5 second intro to the DependencyModel library is that it is the managed object model for the `.deps.json` file that is created as part of every .NET Core application.  The `dotnet.exe` native host uses the .deps.json file to know which assemblies your app requires and which it should load.  Managed code can also retrieve this information to know about the dependencies of the application.

Currently, I'm aware of 2 "high profile" places that uses the DependencyModel:
1. [ASP.NET MVC](https://github.com/aspnet/Mvc/blob/c8a1bb69142792d1046044bde7082b4e3ac8ac27/src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultAssemblyPartDiscoveryProvider.cs#L38) apps use it to discover "Controller" types in the application.
1. The [VS test runner for xunit](https://github.com/xunit/xunit/blob/5133b76c1add95351bef471a6900de6155294370/src/xunit.runner.visualstudio/VsTestRunner.cs#L618) uses it to find if the test references any "runner reporter" assemblies.  

Given it is used by things like the ASP.NET MVC runtime, it would be a HUGE breaking change if we ever stopped producing/maintaining this library.  So I wouldn't be worried about the support for it going away anytime soon.  Also, from the way it sounds, you would only be using this library as a stop-gap to use in .NET Core 1.x applications.  Once you are on `netstandard2.0`, you can use `AppDomain.CurrentDomain.GetAssemblies()` again.

> Alternatively I'm not that impressed with using Microsoft.Extensions.DependencyModel either because it is designed for .NET CLI not .NET Core/Standard

> The current master branch of the cli does not appear to have this code anymore.

The DependencyModel library moved "lower" in the stack from the dotnet/cli repo to dotnet/core-setup - the same repo where the `dotnet.exe` host is maintained.  It was recognized that this library is useful at runtime and not just as part of the "tooling".  It is maintained by the same team that maintains the dotnet/corefx and dotnet/coreclr repos.

> The other option since .NET Standard 2.0 is bringing AppDomain.CurrentDomain.GetAssemblies() back is to just use reflection to call it since coreclr has always had the APIs by the looks of it (just not exposed by corefx),

Using reflection doesn't work on `netcoreapp1.x` because the `GetAssemblies` method was removed in a post-build step since it isn't called internally.  This is why we don't recommend approaches like this.  You can see for yourself if you JustDecompile/ildasm the `System.Private.CoreLib` assembly that ships with `netcoreapp1.x`.  ex. `C:\Program Files\dotnet\shared\Microsoft.NETCore.App\1.1.2\System.Private.CoreLib.dll`.

### Other Thoughts
Using the DependencyModel has an advantage that it gives you a graph of the dependencies.  This is useful if you are searching for types that implement a certain interface, or inherit from a certain class.  You can immediately dismiss any assemblies that don't depend on your core library.  You can see this approach being taken in the ASP.NET MVC example above.  They don't need to inspect assemblies like `Newtonsoft.Json` to see if it contains any MVC Controller classes.  It can't because it doesn't reference the MVC library.

If necessary, you can only depend on the DependencyModel from the `netstandard1.x` build of your project, which means its API usage would be harder to proliferate over time because it would always needed to be `#if` checked.  And when you moved to support `netstandard2.0`, the dependency could be removed, so it would only be supported on the `netstandard1.x` version of your project.